unirec/constants/protocols.py

```diff
@@ -13,6 +13,12 @@ class DataFileFormat(Enum):
     # user_id, item_id.  This is the interaction format.
     T1 = 'user-item' 
 
+    # user_id, item_id, max_len. 
```

Comment:
In fact, the max_len varies for different interactions. This protocol is designed so that the correct subsequence is sampled for each duplicate interaction.